node/pkg/dal/collector/collector.go

```diff
@@ -0,0 +1,198 @@
+package collector
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strconv"
+	"sync"
+
+	"bisonai.com/orakl/node/pkg/aggregator"
+	"bisonai.com/orakl/node/pkg/chain/websocketchainreader"
+	"bisonai.com/orakl/node/pkg/common/keys"
+	"bisonai.com/orakl/node/pkg/common/types"
+	dalcommon "bisonai.com/orakl/node/pkg/dal/common"
+	"bisonai.com/orakl/node/pkg/db"
+	errorSentinel "bisonai.com/orakl/node/pkg/error"
+	klaytncommon "github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	DefaultDecimals = "8"
+	GetAllOracles   = "getAllOracles() public view returns (address[] memory)"
+	OracleAdded     = "OracleAdded(address oracle, uint256 expirationTime)"
+)
+
+type Collector struct {
+	IncomingStream  map[int32]chan aggregator.SubmissionData
+	OutgoingStream  map[int32]chan dalcommon.OutgoingSubmissionData
+	Symbols         map[int32]string
+	FeedHashes      map[int32][]byte
+	CachedWhitelist []klaytncommon.Address
+
+	Ctx        context.Context
```

Comment:
is there a specific reason we are storing the `context` in a struct? Apparently it's not a good practice to store context in a struct, rather it should be passed to functions that require it, unless there it's necessary: https://go.dev/blog/context-and-structs

Reply:
I kept ctx to track its running state, 

ctx here is the one which is generated from param passed through `Start` func below.